lib/systems/default.nix

```diff
@@ -49,12 +101,16 @@ rec {
       # Either of these can be losslessly-extracted from `parsed` iff parsing succeeds.
       system = parse.doubleFromSystem final.parsed;
       config = parse.tripleFromSystem final.parsed;
-      # Determine whether we can execute binaries built for the provided platform.
-      canExecute = platform:
-        final.isAndroid == platform.isAndroid &&
-        parse.isCompatible final.parsed.cpu platform.parsed.cpu
-        && final.parsed.kernel == platform.parsed.kernel;
-      isCompatible = _: throw "2022-05-23: isCompatible has been removed in favor of canExecute, refer to the 22.11 changelog for details";
+
+      # 2023-06-17: `platform.canExecute` is deprecated in favor of `lib.systems.canExecute platform`
+      canExecute        = canExecutePlaceholder final;
+
+      # 2023-06-17: `platform.emulatorAvailable` has been removed in favor of `lib.systems.emulatorAvailable platform`
+      emulatorAvailable = emulatorAvailablePlaceholder final;
```

Comment:
These also don't need to be placeholders, right?

Reply:
These were added by @sternenseemann here: https://github.com/NixOS/nixpkgs/pull/238331#pullrequestreview-1485078058 in commit 5d8e668d16f7ce7c32399007d3a49a27322ac3ed

And yes they do need to be placeholders, because they are functions.  And Nix's definition of equality for functions is crackheaded.  Which is why we are trying so desperately to remove the functions!

Reply:
See https://github.com/NixOS/nixpkgs/commit/5d8e668d16f7ce7c32399007d3a49a27322ac3ed#diff-2165823a8d82c5dd1353601bd290df8bd431f9ee2096750d9ef655cf5d251998R37-R47

```nix
 # These definitions have to be floated out into a `let … in` binding to ensure
 # that they are the same exact value struct in the evaluating Nix
 # implementation. Due to a quirk in how equality is implemented, values including
 # functions are considered equal if they have the same memory location as long as
 # they are inside a container value and not compared directly. Thus any attribute
 # set will compare equal to itself even if it contains functions.
 # By floating the functions out of the attribute set construction we'll ensure
 # that the functions have the same memory location regardless of what call to
 # lib.systems.elaborate produced them. This ensures that platform sets will be
 # equal even if they have been constructed by different calls to lib.systems.elaborate
 # —as long as their other contents are the same.
```

Reply:
Ugh, right.